### ISSUE TITLE
AVD emulator starts but returns "The analysis hit the critical timeout, terminating"

### ISSUE THREAD

**Issue by wickywaka** [idanr1986/cuckoo-droid#50]
Here is the log file generated (result server received messages):

```
2017-10-24 15:31:24,181 [root] INFO: Starting analyzer from: /data/local/tmp/urjtjqy
2017-10-24 15:31:24,184 [root] INFO: Storing results at: /data/local/tmp/ckhici
2017-10-24 15:31:24,186 [root] INFO: Target is: /data/local/tmp/RGB.apk
2017-10-24 15:31:24,187 [root] INFO: No analysis package specified, trying to detect it automagically
2017-10-24 15:31:24,189 [root] INFO: Automatically selected analysis package "apk"
2017-10-24 15:31:24,813 [root] INFO: Started auxiliary module FileCollector
2017-10-24 15:31:24,831 [root] INFO: Started auxiliary module Screenshots
2017-10-24 15:31:24,836 [root] INFO: Started auxiliary module TouchSimulator
2017-10-24 15:31:24,915 [root] INFO: installing sample on emulator: pm install /data/local/tmp/RGB.apk
2017-10-24 15:31:27,077 [root] INFO: 
2017-10-24 15:31:27,081 [root] INFO: finished
2017-10-24 15:31:27,083 [root] INFO: executing sample on emulator:adb shell am start -n com.fastebro.androidrgbtool/com.fastebro.androidrgbtool.ui.MainActivity
```

Only the emulator is launched and nothing else happens.

Here is the traffic of adb port and screenshot from wireshark(Just to have idea of times these commands were exectued):

```
Android Console: Authentication required
Android Console: type 'auth <auth_token>' to authenticate
Android Console: you can find your <auth_token> in 
'/home/wra/.emulator_console_auth_token'
OK
auth V14E+IqrFm78AYgS
kill
quit
Android Console: type 'help' for a list of commands
OK
OK: killing emulator, bye bye
OK
```
![image](https://user-images.githubusercontent.com/6962179/31946279-b4bdfe98-b8d1-11e7-89ab-2f3b2ceca3c6.png)

I am not sure how the app is run, from the result server log it seems like the app is being run using adb, but I don't see anything wireshark captures of adb.

**Comment by wickywaka** [comment]
I solved the bug and its been repeated in several other places. Below are the details (also check #53)

The problem was with `subprocess.Popen()`. By default it looks for shell in /bin/sh but avd it is /system/bin/sh. Have a look here at [documentation](https://docs.python.org/3/library/subprocess.html)
This can be fixed by making minor changes in the command:
1. Make command a string(not a list)
2. Add parameters `shell = True` and `executable = /system/bin/sh` to `subproccess.Popen()`

Disclaimer: Its very well possible that my shell is not located in the right place because of some changes that I made so please verify the location of your shell before making these changes.

**Comment by wickywaka** [comment]
If the above the changes solved your problem, please confirm it in #53
